packages/compass-welcome/src/components/web-welcome-tab.tsx

```diff
@@ -26,8 +33,229 @@ const contentBodyStyles = css({
   alignItems: 'flex-start',
 });
 
+const containerStyles = css({
+  display: 'flex',
+  alignItems: 'flex-start',
+  justifyContent: 'center',
+  margin: '0 auto',
+  minHeight: '100vh',
+  paddingTop: spacing[800], // Move content up with top padding
+});
+
+const connectingLayoutStyles = css({
+  display: 'flex',
+  flexDirection: 'column',
+  alignItems: 'flex-start', // Changed from 'center' to 'flex-start' for left alignment
+  gap: spacing[300], // Reduce gap to compress layout
+  textAlign: 'left', // Changed from 'center' to 'left'
+  maxWidth: '600px',
+  width: '100%',
+});
+
+const connectingImageContainerStyles = css({
+  display: 'flex',
+  justifyContent: 'flex-start', // Align with the text
+  marginBottom: spacing[200], // Add margin below image
+  width: 'fit-content', // Only take up as much width as needed
+  alignSelf: 'flex-start', // Align to the start of the container
+});
+
+const connectingImageStyles = css({
+  width: '334px', // Match the exact text width (333.22px rounded up)
+  height: 'auto', // Maintain aspect ratio
+  maxWidth: '100%', // Ensure it doesn't overflow container
+});
+
+const connectingTitleStyles = css({
+  width: 'fit-content', // Let the title determine its natural width
+  margin: 0, // Remove any default margins
+});
+
+const connectionListStyles = css({
+  display: 'flex',
+  flexDirection: 'column',
+  gap: spacing[300], // Increase gap for better separation of connections
+  alignItems: 'flex-start', // Changed from 'center' to 'flex-start' for left alignment
+  width: '100%',
+  marginTop: spacing[400], // Add more space above connection list
+});
+
+const connectingConnectionItemStyles = css({
+  display: 'flex',
+  alignItems: 'center',
+  gap: spacing[200],
+  fontSize: '14px',
+  color: palette.gray.dark1, // Grayish color for connecting state
+});
+
+const connectedConnectionItemStyles = css({
+  display: 'flex',
+  alignItems: 'center',
+  gap: spacing[200],
+  fontSize: '14px',
+  color: palette.black, // Black color for connected state
+});
+
+const greenIconStyles = css({
+  color: palette.green.dark2, // Use darker green
+});
+
+const subItemStyles = css({
+  display: 'flex',
+  alignItems: 'center',
+  gap: spacing[200],
+  fontSize: '12px',
+  paddingLeft: spacing[600], // Indent for sub-items
+  paddingTop: spacing[100], // Add vertical spacing
+  paddingBottom: spacing[100],
+  color: palette.gray.dark1,
+});
+
+function ConnectionStatus({
+  connectionId,
+  isConnected,
+}: {
+  connectionId: string;
+  isConnected: boolean;
+}) {
+  const connectionInfo = useConnectionInfoForId(connectionId);
+  const connectingSteps = useConnectionConnectingSteps(connectionId);
+
+  if (!connectionInfo) {
+    return null;
+  }
+
+  const connectionName = connectionInfo.title;
+
+  // Create sub-items for connecting steps
+  const renderConnectingSteps = () => {
```

Comment:
A cleaner approach would be to define either this or even a single step outside of the render as a separate component (["Thinking in React"](https://react.dev/learn/thinking-in-react) is an easy read to get a feeling for how to break down code like that)